query-node/substrate-query-framework/cli/src/helpers/SchemaParser.ts

```diff
@@ -30,8 +31,10 @@ export class GraphQLSchemaParser {
     if (!fs.existsSync(schemaPath)) {
       throw new Error('Schema not found');
     }
-    const docNode = parse(fs.readFileSync(schemaPath, 'utf8'));
-    const schema = buildASTSchema(docNode);
+    const ast = parse(fs.readFileSync(schemaPath, 'utf8'));
+    // in order to build AST with undeclared directive, we need to 
+    // switch off SDL validation
+    const schema = buildASTSchema(ast, { assumeValidSDL : true });
```

Comment:
Is it good to assume the schema is valid? I mean one can do a mistake then we will not generate code correctly perhaps there will be errors but not sure...
For custom scalar types (e.g BigInt) I also need to add scalar type definition to the schema (e.g `scalar BigInt`) but setting this to `true` doesn't work for scalar types it is only for directives I think. So I have to do something like this:

```ts
const SCALAR_TYPE_DEFINATIONS = `
scalar BigInt
scalar BigDecimal

// Maybe you can add directive here
// directive @@fullTextSearchable(query: "handles") on FIELD_DEFINITION
`
const ast = parse(SCALAR_TYPE_DEFINATIONS + fs.readFileSync(schemaPath, 'utf8'));
```

Reply:
This is a good point and a possible way to make the SDL valid. There a few things to bear in mind: 1) we then probably need to add the required GraphQL `__Query` top-level entity (as right now we have a rather weird hack that the absence of this top-level entity is _the only_ allowed validation error) 2) we'd still need additional validation checks (like that the directive is applied only on the string fields and also some query-specific ones).

Reply:
That's right!

Reply:
I can add the required change to this PR.

Reply:
That would be really good and I will add my scalar definations then.

Reply:
See the commit below; I had to add another hacky clause to the type name filter so that that bogus `Query` type doesn't show up in the model.